operator/v1/stable.ingresscontroller.testsuite.yaml

```diff
@@ -13,3 +13,84 @@ tests:
       kind: IngressController
       spec:
         httpEmptyRequestsPolicy: Respond
+  - name: Should not allow to set/delete HSTS header.
+    initial: |
+      apiVersion: operator.openshift.io/v1
+      kind: IngressController
+      metadata:
+        name: default-hsts
+        namespace: openshift-ingress-operator
+      spec:
+        httpHeaders:
+          actions:
+            response:
+            - name: X-Frame-Options
+              action:
+                type: Set
+                set:
+                  value: DENY
+            - name: Strict-Transport-Security
+              action:
+                type: Delete
+            request:
+            - name: Content-Location
+              action:
+                type: Set
+                set:
+                  value: /my-first-blog-post
+            - name:   Content-Language
+              action:
+                type: Delete
+    expectedError: "strict-transport-security may not be set via header actions"
+  - name: Should not allow to set/delete dynamic headers with unclosed braces.
+    initial: |
+      apiVersion: operator.openshift.io/v1
+      kind: IngressController
+      metadata:
+        name: default-unclosed-braces
+        namespace: openshift-ingress-operator
+      spec:
+        httpHeaders:
+          actions:
+            response:
+            - name: expires
+              action:
+                type: Set
+                set:
+                  value: "%[date(3600),http_date"
+            request:
+            - name: Content-Location
+              action:
+                type: Set
+                set:
+                  value: /my-first-blog-post
+            - name:   Content-Language
+              action:
+                type: Delete
+    expectedError: "in body should match '^(%(%|(\\{[-+]?[QXE](,[-+]?[QXE])*\\})?([A-Za-z]+|\\[[.0-9A-Z_a-z]+(\\([^)]+\\))?(,[.0-9A-Z_a-z]+(\\([^)]+\\))?)*\\]))|[^%[:cntrl:]])*$'"
```

Comment:
This regex is rough 😅 If you converted the regex to a CEL match regex, you'd be able to provide a better help message, in an ideal world, what would this error message say to help the user to understand the format?

I also wonder, could breaking down the regex into smaller regexes that look at specific subsets of the format, help to given better error messages

Reply:
I am not sure however it is very hard to check each and every part of regex and the provide a message.  Also it should be in one line one is a fetcher then the converter. The person providing dynamic converters should follow a format which is defined in the regex.